apps/hip/forms.py

```diff
@@ -31,3 +35,15 @@ def clean_file(self):
                 except Exception as error:
                     raise forms.ValidationError(error)
             return uploaded_file
+
+
+class HIPAuthenticationForm(AuthenticationForm):
+    def confirm_login_allowed(self, user):
+        """SSO users are not allowed to login using the HIPAuthenticationForm."""
+        # Run the AuthenticationForm's confirmation checks.
+        super().confirm_login_allowed(user)
+        # Raise an error if the user is an SSO user.
+        if is_sso_user(user):
```

Comment:
Suggestions rather than a requested change. Should we also see if the user has has 1) a phila email 2) is not an sso user yet. We could raise a validationError prompting the user to check with their IT department about getting an SSO login.

Reply:
Good thinking. `is_sso_user()` returns `True` for any phila.gov email address, so this should already be handled, since the only way to get an account is to have someone else create one.